backend/geonature/tests/test_synthese.py

```diff
@@ -338,24 +338,44 @@ def test_get_bbox_id_dataset(self, synthese_data, datasets, unexisted_id):
                                          query_string={'id_dataset': unexisted_id})
         assert response_empty.status_code == 204
         assert response_empty.get_data(as_text=True) == ''
-    
+
     def test_observation_count_per_column(self, synthese_data):
         column_name_dataset = 'id_dataset'
         column_name_cd_nom = 'cd_nom'
 
         response_dataset = self.client.get(url_for('gn_synthese.observation_count_per_column', column=column_name_dataset))
         response_cd_nom = self.client.get(url_for('gn_synthese.observation_count_per_column', column=column_name_cd_nom))
 
-        id_datasets = [synt.id_dataset for synt in synthese_data]
-        id_dataset_set = set(id_datasets)
+        datasets_count = {}
```

Comment:
La vérification est lourde (for, for, if), il serait beaucoup plus simple est lisible je pense de créer `expected_resp` puis faire juste `assert expected_resp == resp_json`.
Et pour créer `expected_resp`, `groupby` est sans doute pas mal :
```
ds_keyfunc = lambda s: s.id_dataset
expected_resp = [
    {
        'id_dataset': k,
        'count': len(list(g)),
    }
    for k, g in itertools.groupby(sorted(synthese_data, key=ds_keyfunc), key=ds_keyfunc))
]
```
Et pour les `cd_nom`, utiliser `keyfunc = lambda s: s.cd_nom`.

Reply:
Je ne pourrais pas corriger le test avec un `assert expected_resp == resp_json` direct. En effet, quand il y a déjà des données dans la table synthese, `synthese_data` ne les contient pas et je ne peux donc pas construire le `expected_resp`.

Par contre, je pense quand même pouvoir simplifier le test avec ta proposition.